packages/bpk-component-drawer/src/BpkDrawer.js

```diff
@@ -0,0 +1,78 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { Portal } from 'bpk-react-utils';
+import { withScrim } from 'bpk-scrim-utils';
+
+import BpkDrawerContent from './BpkDrawerContent';
+
+const ScrimBpkDrawerContent = withScrim(BpkDrawerContent);
+
+class BpkDrawer extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      show: true,
+    };
+
+    this.onExited = this.onExited.bind(this);
+    this.hide = this.hide.bind(this);
+  }
+  onExited() {
+    this.props.onClose();
+    this.setState({ show: true }); // needed to show the drawer if it is reopened
+  }
+
+  hide() {
+    this.setState({ show: false });
+  }
+
+  render() {
+    const {
+      isOpen, onClose, target, ...rest
+    } = this.props;
+
+    const { show } = this.state;
+
+    return (
+      <Portal isOpen={isOpen} onClose={this.hide} target={target}>
+        <ScrimBpkDrawerContent
+          show={show}
+          onClose={this.hide}
+          onExited={this.onExited}
+          {...rest}
+        />
+      </Portal>
+    );
+  }
+}
+
+BpkDrawer.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
```

Comment:
I'm unclear as to the difference between the `show` boolean and `isOpen` boolean. Is this one related to whether it is initially shown?

Reply:
Yup, a better name would be `isDrawerShown`.

Reply:
Ah I see - sorry for the misunderstanding. I think `isOpen` would be best as this is consistent with our modals

Reply:
@georgegillams I've pushed a commit that hopefully makes things clearer.

Reply:
Awesome 🎉